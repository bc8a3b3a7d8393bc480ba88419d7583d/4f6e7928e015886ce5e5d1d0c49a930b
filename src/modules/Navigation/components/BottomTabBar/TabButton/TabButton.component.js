import React, { Component } from 'react';
import { Text, Image, View, TouchableWithoutFeedback } from 'react-native';
import PropTypes from 'prop-types';

import styles from './TabButton.component.styles';
import { NavigationImages } from '../../../../../assets/images';

class TabButton extends Component {
  onPress = () => {
    const { onPress, route } = this.props;
    onPress(route);
  };

  render() {
    const { testID, title, selected, icon } = this.props;
    const titleStyle = [styles.titleText, selected && styles.titleTextSelected];
    const iconName = selected ? `${icon}Selected` : icon;
    return (
      <TouchableWithoutFeedback
        testID={testID}
        onPress={this.onPress}
        style={styles.touchingArea}
      >
        <View style={styles.container}>
          <Image style={styles.icon} source={NavigationImages.tab[iconName]} />
          <Text style={titleStyle}>{title}</Text>
        </View>
      </TouchableWithoutFeedback>
    );
  }
}
TabButton.defaultProps = {
  testID: '',
  selected: false,
};

TabButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  testID: PropTypes.string,
  title: PropTypes.string.isRequired,
  selected: PropTypes.bool,
  route: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};

export default TabButton;
