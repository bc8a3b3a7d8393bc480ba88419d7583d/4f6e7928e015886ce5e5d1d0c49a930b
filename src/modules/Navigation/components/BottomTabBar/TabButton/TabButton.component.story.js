import React from 'react';

// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react-native';

import Component from './TabButton.component';

storiesOf('Navigation/Component - TabButton', module)
  .add('default', () => (
    <Component
      title="Tab bar item"
      onPress={() => console.log('pressed button')}
      route="route"
    />
  ))
  .add('selected', () => (
    <Component
      title="Tab bar item"
      onPress={() => console.log('pressed button')}
      route="route"
      selected
    />
  ));
