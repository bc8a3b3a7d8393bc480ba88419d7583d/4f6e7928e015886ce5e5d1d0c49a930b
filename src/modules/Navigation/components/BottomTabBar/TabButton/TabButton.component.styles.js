import { StyleSheet } from 'react-native';
import { COLORS, TEXT_STYLES } from '../../../../../themes';

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  titleText: {
    ...TEXT_STYLES.h4,
    color: COLORS.black300,
  },
  titleTextSelected: {
    color: COLORS.darkBlue,
  },
  icon: {
    width: 20,
    height: 20,
  },
  touchingArea: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
