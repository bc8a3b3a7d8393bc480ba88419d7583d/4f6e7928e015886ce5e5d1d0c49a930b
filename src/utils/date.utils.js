import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
