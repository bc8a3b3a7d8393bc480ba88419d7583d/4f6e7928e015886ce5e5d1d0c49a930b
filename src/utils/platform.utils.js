import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
