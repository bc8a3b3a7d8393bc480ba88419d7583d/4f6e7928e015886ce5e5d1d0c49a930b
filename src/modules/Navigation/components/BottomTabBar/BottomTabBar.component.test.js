import React from 'react';
import { shallow } from 'enzyme';
import { testSnapshots } from '../../../../utils/test.utils';

import Component from './BottomTabBar.component';
import TabButton from './TabButton/TabButton.component';

describe('Navigation/Component - BottomTabBar', () => {
  const componentProps = {
    navigation: {
      navigate: jest.fn(),
    },
  };

  const render = props => shallow(<Component {...props} />);

  testSnapshots(Component, [
    {
      description: 'basic render',
      props: componentProps,
    },
  ]);

  test('should rerender the tab and navigate to selected route after tapping on tab bar item', () => {
    const component = render(componentProps);
    component
      .find(TabButton)
      .at(1)
      .props()
      .onPress('saved');

    const nextRoute = 'saved';
    expect(component.instance().state.selectedTab).toEqual(nextRoute);
    expect(componentProps.navigation.navigate).toBeCalledWith({
      routeName: nextRoute,
    });
  });
});
