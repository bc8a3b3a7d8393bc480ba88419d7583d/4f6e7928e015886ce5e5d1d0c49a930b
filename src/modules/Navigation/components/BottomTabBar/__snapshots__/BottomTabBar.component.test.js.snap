// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation/Component - BottomTabBar snapshots basic render 1`] = `<Component />`;
